src/cli.rs

```diff
@@ -545,19 +545,24 @@ async fn run_deploy(
 
         if deploy_data.merged_settings.interactive_sudo.unwrap_or(false) {
             warn!("Interactive sudo is enabled! Using a sudo password is less secure than correctly configured SSH keys.\nPlease use keys in production environments.");
+            
+            if deploy_data.merged_settings.sudo.is_some() {
+                warn!("Custom sudo commands should be configured to accept password input from stdin when using the 'interactive sudo' option. Deployment may fail if the custom command ignores stdin.");
+            } else {
+                // this configures sudo to hide the password prompt and accept input from stdin
+                if let Some(original) = deploy_defs.sudo.clone() {
```

Comment:
Hmm, I'm not sure if this `if` is needed now. IIUC, `deploy_data.merged_settings.sudo` is `None` at this point and we just simply need to append `-S -p ""` options to it. Am I missing something?

Reply:
When I was testing, given the following node config:
```
nodename = {
        hostname = "1.2.3.4";
        # Use the dynamically derived port number
        sudo = "doas";
        interactiveSudo = true;
        profiles.system = {
          sshUser = "deploy";
          user = "root";
          path = deploy-rs.lib.aarch64-linux.activate.nixos nixosConfigs.nodename;
        };
      };
      ```
      
`deploy_data.merged_settings.sudo` at this point is "doas". if I leave that custom sudo config out of the flake, ie dont define a sudo option,`deploy_data.merged_settings.sudo` is None at the time of this condition. is that intended? If so, it seems like a good way to differentiate when it is safe to automatically append `-S -p ""`, and when it should be left to the user to configure. Happy to do it another way if you suggest

Reply:
Ah. Sorry, I understand now - the innermost `if` should not be needed, but I noticed that when sudo overrides are passed via the cli ie like `--sudo='doas'`, then they do not get merged into the deploy_data.merged_settings. I was going to open another PR as I believe that could be a bug. This innermost `if`, however, covers the case where `deploy_defs` could somehow already have '-S'  in it.